### DISCUSSION TITLE
winter:up - migration error

### DISCUSSION THREAD

**Discussion by rnrkrft** [wintercms/winter#840] [Q&A]
Webinstall and composer run into a MySQL problem: key to short. Tried any suggestion I found so far, but stuck. 
My October sites all use MySQL. So I would like to be able to setup a fresh Winter with MySQL...

**Comment by LukeTowers** [comment]
@rnrkrft please post the full error message you're getting

**Reply by rnrkrft** [reply]
Webinstall stops migration with sql error "key length 1071 not allowed". 
Had to change database to utf8 to install via composer.

I send a screenshot tomorrow

**Reply by rnrkrft** [reply]
![image](https://user-images.githubusercontent.com/6118278/220157285-793103ab-26a9-4769-b128-bfb46614c2f6.png)

**Reply by rnrkrft** [reply]
![image](https://user-images.githubusercontent.com/6118278/220176646-5b48fa4d-62df-4c04-b0d3-2ea6b44378ac.png)![image](https://user-images.githubusercontent.com/6118278/220176727-cb4a4a48-40f3-4281-b8bf-df8f820b6510.png)

**Reply by LukeTowers** [reply]
Sounds like you need to update the version of your database server

**Reply by rnrkrft** [reply]
That's mariadb on a CentOS7 with Plesk 18.0.50 to manage one of the virtual server. I went through the suggestions coming from Plesk community and found this error quite common. Solution for me was to use UTF8 instead UTF8mb4 with the composer setup and the migration (winter:up) runs smooth. 
It is Plesk. I update to mariadb 10.5, but Plesk still uses 5.5.68.

**Reply by LukeTowers** [reply]
@rnrkrft okay, just be aware that using utf8 instead of ut8mb4 will cause issues if you have content stored in the database that contains either emojis or unicode characters from other languages like Chinese or Japanese. I would recommend instead that you find out how to switch Plesk to use the newer database engine (because there will be other incompatibilities that you will encounter and they will be frustrating to debug) and then reinstall using utf8mb4 instead of just utf8.

**Reply by rnrkrft** [reply]
That's the plan, because I ran into the next mysql problem at once. Installed adminer and winter crashes.  I will use fresh install and try again to migrate the October sites on a development server.
